core/dbt/parser/unit_tests.py

```diff
@@ -438,6 +437,31 @@ def process_models_for_unit_test(
     target_model_id = unit_test_def.depends_on.nodes[0]
     target_model = manifest.nodes[target_model_id]
     assert isinstance(target_model, ModelNode)
+
+    target_model_is_incremental = "macro.dbt.is_incremental" in target_model.depends_on.macros
+    unit_test_def_has_incremental_override = unit_test_def.overrides and isinstance(
+        unit_test_def.overrides.macros.get("is_incremental"), bool
+    )
+
+    if target_model_is_incremental and (not unit_test_def_has_incremental_override):
+        raise ParsingError(
+            f"Boolean override for 'is_incremental' must be provided for unit testing model '{target_model.name}'"
+        )
+
+    unit_test_def_incremental_override_true = (
```

Comment:
do we actually need this in addition to `unit_test_def_has_incremental_override`? given we already have `unit_test_def_has_incremental_override` + are past the parsing error above, I think they are equivalent

Reply:
Yes, since we want to test that the value of `unit_test_def_has_incremental_override` is explicitly `True` here.

Above, we're only testing that its value is not `None`.

We don't want this branch to be triggered if the value is `False`.

Reply:
ahh gotcha! thank you